internal/provider/kubernetes/status_updater.go

```diff
@@ -56,14 +57,25 @@ func (m MutatorFunc) Mutate(old client.Object) client.Object {
 type UpdateHandler struct {
 	log           logr.Logger
 	client        client.Client
+	sendUpdates   chan struct{}
 	updateChannel chan Update
+	writer        *UpdateWriter
 }
 
 func NewUpdateHandler(log logr.Logger, client client.Client) *UpdateHandler {
+	sendUpdates := make(chan struct{})
+	updateChannel := make(chan Update, 100)
 	return &UpdateHandler{
 		log:           log,
 		client:        client,
-		updateChannel: make(chan Update, 100),
+		sendUpdates:   sendUpdates,
+		updateChannel: updateChannel,
+		writer: &UpdateWriter{
+			log:                 log,
+			enabled:             sendUpdates,
+			updateChannel:       updateChannel,
+			eventsBeforeEnabled: make(chan Update, 1000),
```

Comment:
isnt this going to have the same issue of missing events before its set ?

Reply:
should the status updater start here https://github.com/envoyproxy/gateway/blob/29b5eaa6a209adb4e336b3c773dc92bdb647c891/internal/provider/kubernetes/controller.go#L132 ?

Reply:
> isnt this going to have the same issue of missing events before its set ?

No, because the `eventsBeforeEnabled` receives updates before the status updater starts.

> should the status updater start here

The status updater must be added to the controller manager and started with it because it needs leader election.